sniproxy.cpp

```diff
@@ -190,7 +190,40 @@ SNIProxy::~SNIProxy()
 
 void SNIProxy::update()
 {
-    m_pixmap = QPixmap::fromImage(getImageNonComposite());
+    QImage newImage = getImageNonComposite();
+
+    if (m_pixmap.isNull()) {
+        m_pixmap = QPixmap::fromImage(newImage);
+    }
+    else if (!newImage.isNull()) {
+        auto isTransparentImage = [](const QImage &image) -> bool {
+            unsigned long long sum = 0;
+
+            for (int x = 0; x < image.width(); ++x) {
+                for (int y = 0; y < image.width(); ++y) {
+                    sum += qAlpha(image.pixel(x, y));
+                }
+            }
+
+            if (sum < 255) {
```

Comment:
it'd be faster if you reversed the logic; 
break the loop if sum ever got bigger than 255